### PR TITLE
UIAccessibility usability improvements for VoiceOver

### PR DESCRIPTION
**Pull request by mathewa6**
Thanks for the library! I came across it on Twitter after it was listed in the top downloaded pods.

Since most of this project is based on UIKit elements, it is accessible. However, there are some usability issues that get in the way of a VoiceOver (VO) user being aware of what is happening on screen.

1) When the HUD is shown, there is no indication given to a VO user unless they swipe.
2) When they do swipe, the list of elements from the HUD spill onto elements from underneath the HUG.
3) Occasionally, the dismissal of the HUD refocuses VO to the top of the superview, which isn't ideal.

This PR attempts to address these issues in a non intrusive manner. All tests passing on my end.
These changes do the following:
- Focus on the HUD when presented and return focus to the last active element after hiding. 
- Prevent VO from scrolling onto elements behind the HUD while it is in progress/animating.

Happy to make changes as needed.

**TLDR**: [Heres a 20s video(no audio)](https://streamable.com/48wcm): Just pay attention to the VO focus box.

### CODE REVIEW

**Comment by matej** [comment]
👏 Thank you for taking interest in this. I'm putting this on my TODO list for testing. Thanks!

**Comment by ay8s** [comment]
This is definitely an improvement. Awesome work @mathewa6! 👍 

I do see the focus switch between the HUD and the previously select element midway through progress. https://cl.ly/32ba0000d2e0 Would also be great to read the label if the indicator has one.

**Comment by mathewa6** [comment]
Thanks. I had to find my notes from when I worked on this to figure out why I hadn't read the label automatically. It seems like I couldn't simply set the accessibilityLabel on the UIActivityIndicatorView to use the self.label text for some reason. (Maybe I'm missing something?).

The other option would be to set the entire MBProgressHUD to be its own accessibilityElement whose label and traits are changed depending on the internal classes used, which I thought is best saved for a bigger PR.

**Comment by matej** [comment]
Took a closer look at this. While I believe that making the whole indicator a single accessibility element would be a nicer solution, I still think this is a step in the right direction and improves the behavior over the current one. One problem is that "Progress" isn't really localized. Taking the label value, when set, would make that better, as I would assume the host app would already make sure the label text is localized.

File: MBProgressHUD.m [hunk 1]
```diff
@@ -1002,6 +1058,8 @@ - (void)drawRect:(CGRect)rect {
     CGContextSetStrokeColorWithColor(context,[_lineColor CGColor]);
     CGContextSetFillColorWithColor(context, [_progressRemainingColor CGColor]);
     
+    self.accessibilityFrame = [self convertRect: rect toCoordinateSpace:[[UIScreen mainScreen] coordinateSpace]];
```
Comment (matej) [inline MBProgressHUD.m]: Spacing nitpick. 

```suggestion
    self.accessibilityFrame = [self convertRect:rect toCoordinateSpace:[[UIScreen mainScreen] coordinateSpace]];
```

File: MBProgressHUD.m [hunk 2]
```diff
@@ -824,6 +858,11 @@ @implementation MBRoundProgressView
 #pragma mark - Lifecycle
 
 - (id)init {
+    // Ensure that this is an accessibility element and set the trait to allow percentage completion to be accessible.
+    self.isAccessibilityElement = YES;
+    self.accessibilityLabel = @"Progress";
```
Comment (matej) [inline MBProgressHUD.m]: Wrapping with `NSLocalizedString` would make it easier for the host app to provide localizations here. 

```suggestion
    self.accessibilityLabel = NSLocalizedString(@"Progress");
```

File: MBProgressHUD.m [hunk 3]
```diff
@@ -945,6 +990,11 @@ @implementation MBBarProgressView
...
+    self.accessibilityTraits = UIAccessibilityTraitUpdatesFrequently;
```
Comment (matej) [inline MBProgressHUD.m]: This is redundant as we're calling through to `initWithFrame:` below which does the same.

**Comment by mathewa6** [comment]
Thanks @matej !

I've made 2 additional commits, the first of which directly addresses the feedback you've posted. 

The second uses KVO on the label's text property and then uses that as accessibilityLabel on the indicator. I've hence removed label as an accessibilityLabel except for text only and custom modes to prevent redundant VoiceOver speech. Do note that this does result in an additional call to  `-updateIndicators`: one at init and one if `self.label.text` is set. Not sure if this is the best way to go about it, but happy to make changes as it's the closest to just making the entire HUD a single element.
